MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.dev.cc

```diff
@@ -0,0 +1,141 @@
+#include <alpaka/alpaka.hpp>
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalVector.h"
+#include "DataFormats/FWLite/interface/Record.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "CLHEP/Geometry/Vector3D.h"
+
+#include "MagneticField/ParametrizedEngine/interface/alpaka/ParabolicParametrizedMagneticField.h"
+
+#include "FWCore/Utilities/interface/stringize.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+#include <iostream>
+#include <fstream>
+#include <Eigen/Core>
+
+using namespace edm;
+using namespace std;
+using namespace ALPAKA_ACCELERATOR_NAMESPACE;
+using namespace ALPAKA_ACCELERATOR_NAMESPACE::magneticFieldParabolicPortable;
+using Vector3f = Eigen::Matrix<float, 3, 1>;
+using Vec3 = CLHEP::Hep3Vector;
+
+struct MagneticFieldKernel {
+  template <typename TAcc, typename T>
+  ALPAKA_FN_ACC void operator()(TAcc const& acc,
+                                Vector3f const* __restrict__ in,
+                                T* __restrict__ out,
+                                size_t size) const {
+    for (auto index : cms::alpakatools::uniform_elements(acc, size)) {
+      out[index*3 + 0] = 0;
+      out[index*3 + 1] = 0;
+      out[index*3 + 2] = magneticFieldAtPoint(in[index]);
+    }
+  }
+};
+
+int main() {
+  // get the list of devices on the current platform
+  auto const& devices = cms::alpakatools::devices<Platform>();
+  if (devices.empty()) {
+    std::cerr << "No devices available for the " EDM_STRINGIZE(ALPAKA_ACCELERATOR_NAMESPACE) " backend, "
+      "the test will be skipped.\n";
+    exit(EXIT_FAILURE);
+  }
+
+  ifstream file;
+  edm::FileInPath mydata("MagneticField/Engine/data/Regression/referenceField_160812_RII_3_8T.bin");
+  file.open(mydata.fullPath().c_str(), ios::binary);
+
+  int count = 0;
+  float px, py, pz;
+  float bx, by, bz;
+  vector<Vector3f> points;
+  vector<GlobalVector> referenceB_vec;
+
+  int numberOfPoints = 100;
+  do {
+    if (!(file.read((char*)&px, sizeof(float)) && file.read((char*)&py, sizeof(float)) &&
+          file.read((char*)&pz, sizeof(float)) && file.read((char*)&bx, sizeof(float)) &&
+          file.read((char*)&by, sizeof(float)) && file.read((char*)&bz, sizeof(float))))
+      break;
+
+    const auto point = Vec3(px, py, pz);
+    if (!isValid(point))
```

Comment:
Is this used anywhere else ?
If not, can use a `Vector3f` like the rest of the code, and remove the dependency on CLHEP ?

Reply:
Changed: https://github.com/cms-sw/cmssw/pull/44360/commits/c6936d662ee52343d6446d101f7660a5c8ddffda